platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/location/LocationComponentActivationOptions.java

```diff
@@ -0,0 +1,189 @@
+    this.locationEngineRequest = locationEngineRequest;
+    this.locationComponentOptions = locationComponentOptions;
+    this.styleRes = styleRes;
+    this.useDefaultLocationEngine = useDefaultLocationEngine;
+  }
+
+  /**
+   * Convenience method to retrieve a {@link LocationComponentActivationOptions} object which is ready to build with
+   *
+   * @return a builder object
+   */
+  @NonNull
+  public static Builder builder(@NonNull Context context, @NonNull Style fullyLoadedMapStyle) {
+    return new LocationComponentActivationOptions.Builder(context, fullyLoadedMapStyle);
+  }
+
+  /**
+   * The application's current context
+   *
+   * @return the application's current context
+   */
+  @NonNull
+  public Context context() {
+    return context;
+  }
+
+  /**
+   * The proxy object for current map style. More info at {@link Style}
+   *
+   * @return the map's fully loaded Style object
+   */
+  @NonNull
+  public Style style() {
+    return style;
+  }
+
+  /**
+   * The {@link LocationEngine} which the {@link LocationComponent} should use
+   *
+   * @return the engine, or null if you'd like to only force location updates
+   */
+  @Nullable
+  public LocationEngine locationEngine() {
+    return locationEngine;
+  }
+
+  /**
+   * The location request which the {@link LocationComponent} should use
+   * @return the LocationEngineRequest object
+   */
+  @Nullable
+  public LocationEngineRequest locationEngineRequest() {
+    return locationEngineRequest;
+  }
+
+  /**
+   * A built {@link LocationComponentOptions} object, which holds the various {@link LocationComponent} styling options
+   * @return the options for styling the actual LocationComponent
+   */
+  @Nullable
+  public LocationComponentOptions locationComponentOptions() {
+    return locationComponentOptions;
+  }
+
+  /**
+   * the LocationComponent style resource (e.g. R.style.style_name)
+   *
+   * @return the style resource.
+   */
+  @Nullable
+  public int styleRes() {
+    return styleRes;
+  }
+
+  /**
+   * True if you want to initialize and use the built-in location engine or false if there should be no
+   * location engine initialized
+   * @return whether the default LocationEngine is used
+   */
+  @Nullable
+  public boolean useDefaultLocationEngine() {
+    return useDefaultLocationEngine;
+  }
+
+  /**
+   * Builder class for constructing a new instance of {@link LocationComponentActivationOptions}.
+   *
+   * since 7.2.0
+   */
+  public static class Builder {
+    private Context context;
+    private Style style;
+    private LocationEngine locationEngine;
+    private LocationEngineRequest locationEngineRequest;
+    private LocationComponentOptions locationComponentOptions;
+    private int styleRes;
+    private boolean useDefaultLocationEngine;
+
+    public Builder(Context context, Style style) {
+      this.context = context;
+      this.style = style;
+    }
+
+    public Builder locationEngine(LocationEngine locationEngine) {
```

Comment:
Since we have both an option to pass an engine and a `useDefaultLocationEngine` flag in one place now, we need to document more clearly that `useDefaultLocationEngine` is ignored when a non-null engine is set.

Reply:
Ok, I adjusted to:

```java
    /**
     * Deliver your own {@link LocationEngine} to the LocationComponent.
     * 
     * The true/false 
     * {@link LocationComponentActivationOptions#builder(Context, Style)#useDefaultLocationEngine}
     * activation option is ignored when a non-null {@link LocationEngine} is set via
     * this `locationEngine()` method.
     *
     * @param locationEngine a {@link LocationEngine} object
     *
     * @return The {@link Builder} class with a set LocationEngine.
     */
    public Builder locationEngine(LocationEngine locationEngine) {
      this.locationEngine = locationEngine;
      return this;
    }
```